l1-contracts-foundry/script/DeployL1.s.sol

```diff
@@ -0,0 +1,710 @@
+        address diamondProxy;
+    }
+
+    struct BridgesDeployedAddresses {
+        address erc20BridgeImplementation;
+        address erc20BridgeProxy;
+        address sharedBridgeImplementation;
+        address sharedBridgeProxy;
+    }
+
+    struct Config {
+        uint256 l1ChainId;
+        uint256 eraChainId;
+        address deployerAddress;
+        address ownerAddress;
+        uint256 gasPrice;
+        ContractsConfig contracts;
+        TokensConfig tokens;
+    }
+
+    struct ContractsConfig {
+        bytes32 create2FactorySalt;
+        address create2FactoryAddr;
+        address multicall3Addr;
+        uint256 validatorTimelockExecutionDelay;
+        bytes32 genesisRoot;
+        uint256 genesisRollupLeafIndex;
+        bytes32 genesisBatchCommitment;
+        uint256 latestProtocolVersion;
+        bytes32 recursionNodeLevelVkHash;
+        bytes32 recursionLeafLevelVkHash;
+        bytes32 recursionCircuitsSetVksHash;
+        uint256 priorityTxMaxGasLimit;
+        uint256 sharedBridgeUpgradeStorageSwitch;
+        PubdataPricingMode diamondInitPubdataPricingMode;
+        uint256 diamondInitBatchOverheadL1Gas;
+        uint256 diamondInitMaxPubdataPerBatch;
+        uint256 diamondInitMaxL2GasPerBatch;
+        uint256 diamondInitPriorityTxMaxPubdata;
+        uint256 diamondInitMinimalL2GasPrice;
+        address governanceSecurityCouncilAddress;
+        uint256 governanceMinDelay;
+    }
+
+    struct TokensConfig {
+        address tokenWethAddress;
+    }
+
+    Config config;
+    DeployedAddresses addresses;
+
+    function run() public {
+        console.log("Deploying L1 contracts");
+
+        initializeConfig();
+
+        instantiateCreate2Factory();
+        deployIfNeededMulticall3();
+
+        deployVerifier();
+
+        deployDefaultUpgrade();
+        deployGenesisUpgrade();
+        deployValidatorTimelock();
+
+        deployGovernance();
+        deployTransparentProxyAdmin();
+        deployBridgehubContract();
+        deployBlobVersionedHashRetriever();
+        deployStateTransitionManagerContract();
+        setStateTransitionManagerInValidatorTimelock();
+
+        deployDiamondProxy();
+
+        deploySharedBridgeContracts();
+        deployErc20BridgeImplementation();
+        deployErc20BridgeProxy();
+        updateSharedBridge();
+
+        updateOwners();
+
+        saveOutput();
+    }
+
+    function initializeConfig() internal {
+        string memory root = vm.projectRoot();
+        string memory path = string.concat(root, "/script-config/config-deploy-l1.toml");
+        string memory toml = vm.readFile(path);
+
+        config.l1ChainId = block.chainid;
+        config.deployerAddress = msg.sender;
+
+        // Config file must be parsed key by key, otherwise values returned
+        // are parsed alfabetically and not by key.
+        // https://book.getfoundry.sh/cheatcodes/parse-toml
+        config.gasPrice = toml.readUint("$.gas_price");
```

Comment:
You still use gas price from config, let's move keep use the foundry for it

Reply:
Done.